Update release process docs (#953)
mattermost/mattermost-handbook: operations/research-and-development/product/release-process/release-overview.md

6. **Release new products tightly integrated with the Mattermost suite.**
   * E.g. Boards will ship as a plugin in June; Playbooks is a plugin. Plugins are the strategy for now, maybe there will be another strategy for the long-term.

## Multi-product Release Processes
 - The sample [Plugin Release Playbook](https://community.mattermost.com/playbooks/playbooks/f4oh16ardfbyfgkas1cb6intmw/preview) helps give an overview of needed steps for plugin releases.
 - Please also refer to the [Playbooks release documentation](https://handbook.mattermost.com/operations/research-and-development/product/release-process/playbooks-release) and [Boards release documentation](https://handbook.mattermost.com/operations/research-and-development/product/release-process/boards-release).

## Overview of Release Cycles

Currently Mattermost Cloud releases occur on a biweekly cycle, but the goal is to release more frequently. Cloud and self-managed PRs may need to be cherry-picked to the release branches, such as regression bug fixes and hotfixes that get merged after the release branch has already been cut.

Schedule for Mattermost Cloud releases:

> * \(T-6\): 7 working days prior to the release day, the `master` branch is merged to the Cloud branch and Cloud test servers are updated.

Reviewer: wondering what is the reason to merge so early

Author: QA wants to have test servers updated by Friday morning (European time) latest, so that they have enough time to run automated tests and smoke tests.

Ibrahim completes the "fast-forward" on Wednesday or Thursday, and test servers are updated on Thursday or Friday morning.

Reviewer: ok I see @amyblais thanks for the details

**Understanding the cadence**

> * The Mattermost Cloud releases follows a biweekly cycle and the release day is normally on Wednesdays. Feature Complete deadline for each Mattermost Cloud release is on Mondays 7 working days prior to the Cloud release day.

Reviewer 2: ```suggestion
* The Mattermost Cloud releases follow a biweekly cycle and the release day is normally on Wednesdays. Feature Complete deadline for each Mattermost Cloud release is on Mondays 7 working days prior to the Cloud release day.
```

* A: Desktop releases are currently released as required.

**Q When do I need to have a feature PR to be included into the next release?**

> * A: Aim to have the PR merged before the feature complete deadline. The earlier in the monthly cycle the PR is merged, the higher the chances are for it to be included in that months release. Quality of our releases is important and feature PRs are normally never cherry-picked to a release branch.

Reviewer 2: ```suggestion
* A: Aim to have the PR merged before the Feature Complete deadline. The earlier in the monthly cycle the PR is merged, the higher the chances are for it to be included in that month's release. The quality of our releases is important and feature PRs are not normally cherry-picked to a release branch.
```

**Q: Will RC testing remain, or will RC testing process change?**

* There is still manual QA RC testing for Mobile App and self-managed releases until all release tests are automated.

**Q: Do Mobile and Desktop App releases follow the "cloud first" strategy?**

> * Mobile and Desktop App releases are not based off of Mattermost Cloud releases at this point. These releases are based off of ``master``.

Reviewer 2: ```suggestion
* Mobile and Desktop App releases are not based off of Mattermost Cloud releases at this point. These releases are based off of `master`.
```
